### ISSUE TITLE
Floating-point errors causing NBT mismatches between client and server

### ISSUE THREAD

**Issue by dktapps** [pmmp/PocketMine-MP#4061]
### Issue description
Since the server doesn't keep track of the serial network NBT of items anymore, it's possible for floating-point errors to slightly alter the decoded result, which leads to a false negative comparison.

This is problematic because it leads to items not being considered equivalent even though they are, for all intents and purposes, the same item. This causes problems with inventory transactions, amongst other things.

### Steps to reproduce the issue
1. Use the following code in a plugin
```php
	public function onPlayerChat(PlayerChatEvent $event) : void{
		$event->getPlayer()->getInventory()->setItem($event->getPlayer()->getInventory()->getHeldItemIndex(), VanillaItems::LAPIS_LAZULI()->setNamedTag(
			CompoundTag::create()->setFloat("myfloat", 0.3)
		));
	}
```
2. Chat (you'll get a lapis-lazuli item)
3. Try to move the item in your inventory. You'll notice it doesn't want to move.

### OS and versions

* PocketMine-MP: 3.17.5 & 4.0.0
* PHP: 7.4.15

**Comment by dktapps** [comment]
Fixed by 712df04bc4710ee5e9585bf5a6e7e6fbc772a2ad.

**Comment by SOF3** [comment]
(The actual fix is in https://github.com/pmmp/NBT/commit/7de6bdfdfbcafe2689aafbde47f6506f265afd19)
